### ISSUE TITLE
v4.0.1 fails to build on OS X 10.11.6

### ISSUE THREAD

**Issue by leonklingele** [open-mpi/ompi#6583]
```sh
$ brew upgrade open-mpi
[..]
==> make all
Last 15 lines from /Users/user/Library/Logs/Homebrew/open-mpi/02.make:
/bin/sh ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I../src/include -I../include   -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/mca/pmix/pmix3x/pmix -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/mca/pmix/pmix3x/pmix/src -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/mca/pmix/pmix3x/pmix/include -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1 -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/include -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/mca/hwloc/hwloc201/hwloc/include -I/usr/local/opt/libevent/include  -DNDEBUG -O3 -fno-strict-aliasing  -c -o runtime/pmix_finalize.lo runtime/pmix_finalize.c
/bin/sh ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I../src/include -I../include   -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/mca/pmix/pmix3x/pmix -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/mca/pmix/pmix3x/pmix/src -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/mca/pmix/pmix3x/pmix/include -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1 -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/include -I/private/tmp/open-mpi-20190412-69028-1avxqe7/openmpi-4.0.1/opal/mca/hwloc/hwloc201/hwloc/include -I/usr/local/opt/libevent/include  -DNDEBUG -O3 -fno-strict-aliasing  -c -o runtime/pmix_init.lo runtime/pmix_init.c
server/pmix_server_ops.c:3432:19: error: use of undeclared identifier 'CLOCK_MONOTONIC'
    clock_gettime(CLOCK_MONOTONIC, &tp);
                  ^
1 error generated.
make[5]: *** [server/pmix_server_ops.lo] Error 1
make[5]: *** Waiting for unfinished jobs....
[..]
```

Clang version:

```sh
$ clang -v
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by jsquyres** [comment]
That's a bit odd, because the v4.0.1 tarball builds fine for me on MacOS 10.14.3 outside of homebrew.

I see you're using an older clang -- is this an older MacOS / OS X?

```
$ clang --version
Apple LLVM version 10.0.1 (clang-1001.0.46.3)
Target: x86_64-apple-darwin18.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by leonklingele** [comment]
> is this an older MacOS / OS X

Yep, it's on OS X 10.11.6.

**Comment by jsquyres** [comment]
@rhc54 @ggouaillardet @hppritcha @gpaulsen Does PMIx v3.1.x (bundled in Open MPI 4.0.x) support OS X 10.11 (El Capitan)?

**Comment by rhc54** [comment]
Yes we do - however, there is a patch in master that hasn't been backported to that release branch yet, and so the update didn't get into OMPI 4.0.1

**Comment by jsquyres** [comment]
Ok.  I tagged this as for v4.0.x, but marked it as "minor" because it's such an old version of OS X.

**Comment by ludwigschwardt** [comment]
Thanks for this! I had the same issue on Mac OS X 10.10.5 and `clang`
```
$ clang --version
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.5.0
Thread model: posix
```
Once I built the master branch (via `brew install --HEAD openmpi`), all is well.

**Comment by gpaulsen** [comment]
@hppritcha has verified that this is fixed on v4.0.x and will be fixed in v4.0.2.

**Comment by jsquyres** [comment]
v4.0.2 has been released; I think this has been fixed.  Closing.
